### PR TITLE
[20908] new table component

### PR DESCRIPTION
**Pull request by HDinger**
started adapting the new table component to the required design

https://community.openproject.org/work_packages/20908

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hdinger-fix-20908-new-table-component.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hdinger-fix-20908-new-table-component.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
I'd not call the classes `table-component--anything`, because all our components are... components.
I'd rather call it a _table_ (`table--container`, `table--sort-header`).

Or if we want to distinguish this from other tables that there might be (because `table` alone might be too generic), we could try to find a descriptive adjective for this kind of table. But I am really bad at this so my best proposal would be `advanced-table`, which kind of says "I have no idea what makes this special, but it is surely too special to just call it `table`". So probably just `table` is a good choice ^^

File: app/assets/stylesheets/content/_table.lsg [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          <td>Nunc molestie neque sit amet eros semper dapibus.</td>
+        </tr>
+        <tr>
+          <td>Nullam a sem et metus congue placerat.</td>
+          <td>Mauris ut augue viverra, consequat eros eu, maximus quam.</td>
+          <td>Maecenas elementum orci a varius suscipit.</td>
+          <td>Lorem ipsum dolor sit amet, consectetur adipiscing elit</td>
+          <td>Nunc molestie neque sit amet eros semper dapibus.</td>
+        </tr>
+            <tr>
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_table.lsg]: indentation error

**Comment by oliverguenther** [comment]
@NobodysNightmare If there is nothing that makes it _advanced_, what about `generic-table` ? :grinning:

**Comment by NobodysNightmare** [comment]
@oliverguenther Good idea... at least if we find out, that we are afraid of name clashes. Otherwise `table` would be fine ^^

@floriank Probably you are better at reviewing this. This seems to be quite a lot of Sass and there are many ~~hacks~~ things from the WP table that were taken here and I can't judge well, whether this is adequate or not.

**Comment by ulferts** [comment]
@HDinger the PR looks good to me. I added the highlighting functionality in #3333. When this one is merged, it should be possible to add the `highlight-col` attribute to the `col` elements in the LSG. e.g.

```
<colgroup>
 <col highlight-col>
 <col highlight-col>
 <col highlight-col>
 <col highlight-col>
 <col highlight-col>
</colgroup>
```

**Comment by NobodysNightmare** [comment]
@HDinger I merged the PR by @ulferts, you can now merge `dev` into this branch... (or rebase this branch onto `dev`) and go forth as described by @ulferts.

**Pull request merged**